src/components/TransactionList/DetailsModal.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+
+import Grid from "@material-ui/core/Grid/Grid";
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+import Slide from '@material-ui/core/Slide';
+import Button from '@material-ui/core/Button';
+
+import { TransitionProps } from '@material-ui/core/transitions';
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(theme => ({
+    dialogTitle: {
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "space-between",
+        '& h2': {
+            width: "100%",
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "space-between",
+        }
+    },
+    dialogAction: {
+        paddingTop: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+        paddingRight: theme.spacing(3),
+        paddingLeft: theme.spacing(3)
+    }
+}));
+
+const Transition = React.forwardRef<unknown, TransitionProps>(function Transition(props, ref) {
+    return <Slide direction="up" ref={ref} {...props} />;
+});
+
+const tableRowGenerator = (data: any) => {
+    return (
+        Object.keys(data).map(item => {
+            return (<TableRow key={item}>
+                <TableCell>{item}</TableCell>
+                <TableCell>{data[item]}</TableCell>
+            </TableRow>)
+        })
+    );
+};
+
+type Props = {
+    open: boolean,
+    handleClose: any,
+    data: any
+};
+
+const DetailsModal = ({ open, handleClose, data }: Props): React.ReactElement => {
+    const classes = useStyles();
+    return (
+        <div>
+            <Dialog
+                open={open}
+                TransitionComponent={Transition}
+                keepMounted
+                onClose={handleClose}
+                aria-labelledby="alert-dialog-slide-title"
+                aria-describedby="alert-dialog-slide-description"
+            >
+                <DialogTitle className={classes.dialogTitle} onClick={handleClose}>
+                    {"Transaction Details"}
+                    <IconButton onClick={handleClose} color="primary">
+                        <CloseIcon />
```

Comment:
Ja bym nie dawał ikonki `X` i buttona `Close` na raz jeśli oba robią dokładnie to samo (bo to sugeruje że jednak jakaś różnica może być).

Reply:
Cos w tym jest - Jak jest krotki modal, moze rzeczywiscie wydawac sie bez sensu, ale rozne osoby maja rozne przyzwyczajania. 
![image](https://user-images.githubusercontent.com/54579398/63937748-6b2a0600-ca63-11e9-9057-b05c36617fc1.png)

Reply:
👍 a no sensowne wytłumaczenie, osobiście nadal wolał bym 1 żebym nie musiał się zastanawiać które kliknąć i czy to to samo (a jak wiadomo user nie powinien wgl myśleć o takich rzeczech) ale to z nie zmienianiem przyzwyczajeń też ma sens więc faktycznie co kto lubi byle spójnie w obrębie apki 😉(to chyba pierwszy modal tutaj więc możesz zadecydować 😄)